tools/sof-kernel-log-check.sh

```diff
@@ -22,16 +22,19 @@ project_key="sof-audio"
 #echo "run $0 with parameter '$*' for check kernel message error"
 
 if [ "$ignore_str" ]; then
-    err=$(eval $cmd|grep 'Call Trace' -A5 -B3)$(eval $cmd | grep $project_key | grep -E "$err_str"|grep -vE "$ignore_str")
+    err=$(eval "$cmd"|grep 'Call Trace' -A5 -B3)$(eval "$cmd" | grep "$project_key" | grep -E "$err_str"|grep -vE "$ignore_str")
 else
-    err=$(eval $cmd|grep 'Call Trace' -A5 -B3)$(eval $cmd | grep $project_key | grep -E "$err_str")
+    err=$(eval "$cmd"|grep 'Call Trace' -A5 -B3)$(eval "$cmd" | grep "$project_key" | grep -E "$err_str")
 fi
 
 if [ "$err" ]; then
-    echo `date -u '+%Y-%m-%d %T %Z'` "[ERROR]" "Caught dmesg error"
+    echo "$(date -u '+%Y-%m-%d %T %Z')" "[ERROR]" "Caught dmesg error"
     echo "===========================>>"
     echo "$err"
     echo "<<==========================="
+    # exclude none alsa trace, but still print trace message
+    snd_trace=$(echo "$err" | grep "snd")
```

Comment:
I'm afraid here you will ignore 'sof' refer 'snd'

Reply:
for alsa layer, function name is snd_*, for SOC layer, function name is snd_soc_*, and for SOF layer, function name is snd_sof_*, so only grep "snd" should be enough.